sympy/physics/units/prefixes.py

```diff
@@ -90,7 +90,10 @@ def __mul__(self, other):
         return self.scale_factor * other
 
     def __div__(self, other):
-        fact = self.scale_factor / other.scale_factor
+        try:
```

Comment:
`isinstance` would be preferred.

Reply:
We have to use ` if not isinstance(other, Prefix) and not isinstance(other, Quantity):` instead of try-except block to get the desired behaviour. But this is not possible because we can't import Quantity due to circular import problem.  @Upabjojr if I am missing something please point out.

Reply:
The try-except block isn't clear to read. We need to keep the code readable.

An alternative is `if hasattr(other, "scale_factor")`